xbmc/platform/darwin/tvos/MainController.h

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (C) 2010-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#import "windowing/XBMC_events.h"
+
+#import <UIKit/UIKit.h>
+
+typedef enum
+{
+  IOS_PLAYBACK_STOPPED,
+  IOS_PLAYBACK_PAUSED,
+  IOS_PLAYBACK_PLAYING
+} IOSPlaybackState;
+
+typedef NS_ENUM(NSUInteger, UIPanGestureRecognizerDirection)
+{
+  UIPanGestureRecognizerDirectionUndefined,
+  UIPanGestureRecognizerDirectionUp,
+  UIPanGestureRecognizerDirectionDown,
+  UIPanGestureRecognizerDirectionLeft,
+  UIPanGestureRecognizerDirectionRight
+};
+
+@class MainEAGLView;
+
+@interface MainController : UIViewController <UIGestureRecognizerDelegate>
+{
+@private
+  UIWindow *m_window;
+  MainEAGLView *m_glView;
+  // Touch handling
```

Comment:
This comment sounds wrong, as tvOS doesn't run on a touch screen device.

Reply:
The AppleTV Siri remote has a touch pad, so tvOS requires touch handling for it. It's just a different kind of touch :)